General Docs Improvements

gongjason opened the pull request:
refs https://github.com/firezone/product/issues/538

PR with this draft PR as base. Review these first:
- [x] https://github.com/firezone/firezone/pull/1185
- [x] https://github.com/firezone/firezone/pull/1187

Commit:
docs: add logs and update troubleshooting

Updates the troubleshooting doc for Docker based deployments and adds a general logs doc with more info on what Firezone logs.

Also added a mention of Audit logs in beta to source more convos.

Commit:
docs: further Docker clarifications

Clarifying some troubleshooting guides that refer to firezone-ctl.

Not everyone arrives at the env-vars or config file through the deploy/configure guide. Some load these pages directly via search. Adding a more prominent note that it's for Omnibus based deploy.

Commit:
docs: add troubleshooting docs on locked out state

Adding a section on reseting admin passwords for both docker and omnibus deployments. I realize we removed the omnibus firezone-ctl reference. I won't add it back since I have no idea where to put it.

Commit:
docs: update split tunnel doc

Was originally going to write more about other ways you can achieve split tunneling with wireguard, but will leave that for now

Commit:
fix whitespace

Commit:
update split tunnel doc

Commit:
docs: add DNS considerations to split tunnel doc

https://github.com/firezone/product/issues/527

Commit:
update split tunnel doc

Commit:
further clarify split tunnel doc

Commit:
add note about on-demand feature

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+Firezone maintains two types of logs tied to user identity: configuration logs and network activity logs.
+
+Configuration logs track events related to the configuration of Firezone itself, either by admins or users.
+Access logs track connections to protected resources made by users of your network.
```
`Access logs` is used here but `network activity logs` used in other places -- better to be consistent in marketing, docs, etc.

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+* Enable log rotation and max file size
+* [Customize log driver output](https://docs.docker.com/config/containers/logging/log_tags/)
+
+## Audit Logs
```
This section probably needs to be its own page and stay in draft format for the time being. Still working through the architecture and implementation.

gongjason replied:
moved to a draft doc

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+This guide is written for Docker based deployments of Firezone
+:::
+
+## What Firezone Logs
```
Did the capitalization convention change?

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+
+Each container stores logs as a JSON file on the host machine. These files are found at
+`var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log` and can be shown through the
+`docker logs {CONTAINER}` command.
```
This section is applicable only to Linux (and looks to be specific to Ubuntu in particular).

It's probably worth rewriting it using `docker compose logs` instead of `docker logs`. Compose abstracts most of the system-specific parts.

Users need to be aware that `docker compose` commands should either be run by within the Firezone root directory at `$HOME/.firezone` or be run with the `-f $HOME/.firezone/docker-compose.yml` flag so that Docker Compose knows where the compose file is.

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+`var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log` and can be shown through the
+`docker logs {CONTAINER}` command.
+
+## Managing and Configuring Logs
```
Capitalization

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+
+By default, Firezone uses the `json-file` logging driver without
+[additional configuration](https://docs.docker.com/config/containers/logging/json-file/). This means
+logs from each container is stored separately in a file format designed to be exclusively accessed by
```
```suggestion
logs from each container are individually stored in a file format designed to be exclusively accessed by
```

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+or
+[Google Cloud Logging](https://docs.docker.com/config/containers/logging/gcplogs/)
+* Enable log rotation and max file size
+* [Customize log driver output](https://docs.docker.com/config/containers/logging/log_tags/)
```
I think this is pretty neat, and some users will find it useful. Are we advocating users send audit logs (configuration & network activity) to Splunk / GCP as well? Perhaps we should make a clearer distinction.

gongjason replied:
Since we're not 100% sure how audit logs (config + network activity) are generated and stored at the moment, maybe we can address this later. I put the audit logs section in a separate draft doc.

If they're in the database and require different instructions to view + export, we can add instructions in a separate section with some [Admonitions](https://docusaurus.io/docs/markdown-features/admonitions)? 

I renamed "What Firezone logs" to "Deployment logs".

jamilbk replied:
`Infrastructure logs` or `host logs` or `Debug Logs` *may* be a better term. But yeah, they're going to be completely separate from the STDOUT-based logs managed by Docker. They'll hit the database for sure -- they need to be backed up, redundant, stored with some guarantee of integrity, and possible encrypted. All things supported well by `Ecto.Schema` modules.

jamilbk commented on docs/docs/administer/logs.mdx:
```diff
@@ -0,0 +1,60 @@
+Configuration logs track events related to the configuration of Firezone itself, either by admins or users.
+Access logs track connections to protected resources made by users of your network.
+
+Logged events are recorded as JSON objects and stored in the database. These are accessible via a
```
Just want to clarify that the Docker logs described above are not stored in the database. So we have 3 kinds of logs it seems:

* Infrastructure / deployment logs
* Configuration change logs
* Network activity logs

Thoughts on how to segregate or otherwise draw boundaries between these? I foresee confusion on the horizon.

gongjason replied:
Having three separate sections on 1 page could work. I changed "What Firezone logs" to "Deployment logs" in the interim before we add config + network activity logs.

We could use [admonitions](https://docusaurus.io/docs/markdown-features/admonitions) for callouts (i.e. enterprise plan, database vs. json storage)

jamilbk replied:
Respectfully disagree. They're going to differ quite substantially in the following ways:

* What they're used for
* Who they're used by (deployment: ICs, audit: CISO / legal / HR)
* How they're used / viewed / accessed
* The integrity level needed
* Where and how they're stored
* Data retention policies / periods

gongjason replied:
Ok, agreed. Our current logs are more useful for debugging your deployment. Seems more in line with 'troubleshooting' and written for admins.

Let's split out audit logs and put it in 'references' or 'user guides' when we release that feature. It's more of a feature than a deployment detail.  

What do you think?

jamilbk replied:
👍

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
```diff
@@ -108,6 +132,52 @@ To                         Action      From
+## Locked Out of Admin Access
+
+If you are locked out of the admin account for your Firezone instance
+you can re-create your admin user.
```
```suggestion
you can recreate your admin user.
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
````diff
@@ -108,6 +132,52 @@ To                         Action      From
+Run the following command to reset the password on the default admin user.
+
+```shell
+firezone-ctl create-or-reset-admin
````
```suggestion
sudo firezone-ctl create-or-reset-admin
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
````diff
@@ -108,6 +132,52 @@ To                         Action      From
+```
+
+This will reset the password on admin user using the default password
+found in the `.env` environment file.
````
```suggestion
    First change directory to your Firezone root, then run the `bin/create-or-reset-admin` script
    to reset the admin user's password. The password for the user specified by 
  `ADMIN_EMAIL` 
    in `$HOME/.firezone/.env` will be reset to the `DEFAULT_ADMIN_PASSWORD` variable.
  
    ```shell
      cd $HOME/.firezone
      docker compose exec firezone bin/create-or-reset-admin
    ```

    **Note**: If local authentication is disabled, resetting the admin user's password will not 
    re-enable it.
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
````diff
@@ -108,6 +132,52 @@ To                         Action      From
+
+```SQL
+UPDATE configurations SET local_auth_enabled = 't'
+```
````
```suggestion
If issues arise with your identity provider integration, it's possible you could be locked out of the admin portal. To re-enable local authentication so you can log in and resolve the issue, run the following query on the host of your Firezone instance:

   ```shell
   cd $HOME/.firezone
   docker compose exec postgres psql -U postgres -h 127.0.0.1 -d firezone -c "UPDATE configurations SET local_auth_enabled = 't'"
   ```
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -1,22 +1,27 @@
-The `Allowed IPs` field found on the `/settings/default` page determines the IP
-ranges for which the client will route network traffic. Modifications to this
-field will apply only to new WireGuard tunnel configurations generated by Firezone.
+## Step 1 - Set AllowedIPs
```
consistency

```suggestion
## Step 1: Set AllowedIPs
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
 :::
 
-## Step 2 - Regenerate WireGuard configurations
+## Step 2 - Set the DNS (optional)
```
```suggestion
## Step 2: Set the DNS (optional)
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
+over the encrypted VPN tunnel (recommended), ensure the DNS IPs are included in `AllowedIPs`.
+:::
+
+## Step 3 - Regenerate the device configurations
```
```suggestion
## Step 3: Regenerate the device configurations
```

Commit:
Docs - Improve titles and descriptions (#1187)

* docs: update titles, descriptions

Changes:
- set more descriptive titles (better for readers and SEO)
- add meta descriptions
- convert titles to sentence case for consistency

* docs: small fixes

* fix trailing whitespaces

* fix spelling

* Apply suggestions from code review

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: Jason G <jason@firez.one>

* add descriptions to client instructions

Signed-off-by: Jason G <jason@firez.one>
Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

Commit:
chore: update logs doc

Commit:
chore: apply suggestions from code review

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: Jason G <jason@firez.one>

Commit:
chore: line length and log description

Commit:
chore: rename debug logs

jamilbk commented on docs/docs/administer/debug-logs.mdx:
```diff
@@ -0,0 +1,61 @@
+title: Debug Logs
+sidebar_position: 8
+description:
+     Docker deployments of Firezone generate and stores debug logs to a JSON
```
```suggestion
     Docker deployments of Firezone generate and store debug logs to a JSON
```

jamilbk commented on docs/docs/administer/debug-logs.mdx:
```diff
@@ -0,0 +1,61 @@
+This article is written for Docker based deployments of Firezone.
+:::
+
+Docker deployments of Firezone consist of 3 running containers.
```
```suggestion
Docker deployments of Firezone consist of 3 running containers:
```

jamilbk commented on docs/docs/administer/debug-logs.mdx:
```diff
@@ -0,0 +1,61 @@
+`var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log`.
+
+Run the `docker compose logs` command to view the log output from all running
+containers. Note, `docker compose` commands need to run in the Firezone
```
```suggestion
containers. Note, `docker compose` commands need to be run in the Firezone
```

jamilbk commented on docs/docs/user-guides/client-instructions.mdx:
```diff
@@ -79,7 +105,7 @@ ipv6.method: method "auto" is not supported for WireGuard
 :::
 
-### Step 1 - Install the WireGuard Tools
+### Step 1 - Install the WireGuard tools
```
```suggestion
### Step 1: Install the WireGuard tools
```

jamilbk reviewed (changes requested):
Almost there

Commit:
chore: apply suggestions from code review

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: Jason G <jason@firez.one>

Commit:
Docs - Add draft whitelist sso doc (#1185)

* doc: adding whitelist sso doc

redo of https://github.com/firezone/firezone/pull/995

* docs: add seo titles

* docs: improve titles

* grammar fixes

* fix trailing whitespace

apparently my vscode plugin does not find trailing whitespace in front matter

* put doc in draftmode

* add note on wireguard client apps

* Revert "add note on wireguard client apps"

This reverts commit 9013ce1c3b1bcb8b825f0fcbb1c5fe7e050cae82.

* Apply suggestions from code review

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: Jason G <jason@firez.one>

Signed-off-by: Jason G <jason@firez.one>
Co-authored-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk commented on docs/docs/user-guides/client-instructions.mdx:
```diff
@@ -1,14 +1,23 @@
 ---
 
-## Install and Setup
+# End-user Client Instructions
```
Capitalization

jamilbk commented on docs/docs/user-guides/client-instructions.mdx:
```diff
@@ -48,22 +57,39 @@ follow the steps below. You will need:
+
+On the MacOS and Windows desktop apps you can automatically
+enable the VPN on Wi-fi networks and/or Ethernet connections.
+
```
We'll get support questions about this. Maybe better to point them to WireGuard's docs or remove this section.

```suggestion
```

gongjason replied:
I don't think WireGuard keeps any documentation on their client apps. I understand we don't want to maintain any docs for a product we don't have control over. Will remove.

jamilbk commented on docs/docs/user-guides/egress-rules.mdx:
```diff
@@ -1,8 +1,13 @@
+    Firezone's WireGuard®-based secure access platform.
 ---
 
+# Network Access Control
```
Capitalization

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/README.mdx:
```diff
@@ -1,8 +1,11 @@
 ---
 title: NAT Gateway
 sidebar_position: 3
+description: Set a static IP for egress traffic with Firezone's WireGuard® based secure remote access platform.
```
Consistency: `secure remote access` -> `secure access`.

```suggestion
description: Set a static IP for egress traffic with Firezone's WireGuard®-based secure access platform.
```

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/README.mdx:
```diff
@@ -1,8 +1,11 @@
+description: Set a static IP for egress traffic with Firezone's WireGuard® based secure remote access platform.
 ---
 
+# Static egress IP with a NAT Gateway
```
```suggestion
# Static egress IP with a NAT gateway
```

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/README.mdx:
```diff
@@ -31,6 +32,8 @@ through a Firezone server on an EC2 instance. In this case Firezone is acting as
 
+In this example the protected resource and Firezone are in separate VPC regions.
+
 ### Step 1 - Deploy Firezone server
```
```suggestion
### Step 1: Deploy Firezone server
```

jamilbk replied:
Consistency -- probably a good idea to update all of these

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/README.mdx:
```diff
@@ -60,13 +63,9 @@ party to allow traffic from the static IP set in Step 1 (in this case `52.202.88
 
 ### Step 3 - Route traffic to the protected resource through the VPN server
```
```suggestion
### Step 3: Route traffic to the protected resource through the VPN server
```

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx:
```diff
@@ -0,0 +1,87 @@
+---
+title: Restrict SSO to Firezone Users
+sidebar_position: 1
+description: Configure your identity provider to whitelist traffic from Firezone's WireGuard® based secure access platform.
```
```suggestion
description: Configure your identity provider to whitelist traffic from Firezone's WireGuard®-based secure access platform.
```

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx:
```diff
@@ -0,0 +1,87 @@
+
+# Restrict application authentication to VPN users
+
+Some identity providers support restricting SSO functionality
+Some identity providers support restricting SSO access to a predefined CIDR range or list of IP addresses. You can set this to the IP address of
+your Firezone gateway to restrict SSO to users connected to Firezone.
```
Did this draft end up in this PR somehow?

gongjason replied:
https://github.com/firezone/firezone/pull/1185 was merged into this PR. Was trying to consolidate all the docs changes into 1 PR.

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
 :::
 
-## Step 2 - Regenerate WireGuard configurations
+## Step 2: Set the DNS (optional)
```
```suggestion
## Step 2: Set the DNS server(s) (optional)
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
-## Step 2 - Regenerate WireGuard configurations
+## Step 2: Set the DNS (optional)
+
+You can define the DNS used by devices when the WireGuard tunnel
```
```suggestion
You can define the DNS server(s) used by devices when the WireGuard tunnel
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
+
+You can define the DNS used by devices when the WireGuard tunnel
+is active. By default, it uses Cloudflare's DNS servers (`1.1.1.1, 1.0.0.1`).
+Visit the `/settings/default` to override this value.
```
```suggestion
Visit the `/settings/default` page to override this value.
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
+is active. By default, it uses Cloudflare's DNS servers (`1.1.1.1, 1.0.0.1`).
+Visit the `/settings/default` to override this value.
+
+For split tunneling, this can be desired if you run a DNS server that
```
```suggestion
For split tunneling, this may be desired if you run a DNS server that
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
+resolves internal hosts to private IPs reachable via Firezone.
+
+:::note
+If you are using Firezone as a anonymity service or wish to resolve DNS queries
```
```suggestion
If you wish to resolve DNS queries
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
```diff
@@ -1,13 +1,31 @@
 ---
 title: Troubleshoot
 sidebar_position: 6
+description: Troubleshoot common connectivity and configuration issues with Firezone's WireGuard®-based remote access platform.
```
Consistency -- `secure access` in some places but `remote access` in others.

```suggestion
description: Troubleshoot common connectivity and configuration issues with Firezone's WireGuard®-based secure access platform.
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
```diff
@@ -1,13 +1,31 @@
+<Tabs>
+<TabItem value="docker" label="Docker" default>
+
+Each container stores logs as a JSON file on the host machine. These can be shown through the
```
```suggestion
Each container stores logs as a JSON file on the host machine. These can be shown with the
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
```diff
@@ -1,13 +1,31 @@
+
+Each container stores logs as a JSON file on the host machine. These can be shown through the
+`docker logs {CONTAINER}` command. Log files are found at
+`var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log`.
```
```suggestion
`var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log` by default.
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
````diff
@@ -24,7 +42,7 @@ Similarly, to clear SAML configs:
 ```
 
-## Debugging Portal Websocket Connectivity Issues
+## Debugging portal websocket connectivity issues
````
Proper noun

```suggestion
## Debugging portal WebSocket connectivity issues
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
````diff
@@ -108,6 +132,57 @@ To                         Action      From
 51820/udp (v6)             ALLOW IN    Anywhere (v6)
 ```
 
+## Locked out of admin access
````
```suggestion
## Admin login isn't working
```

jamilbk commented on docs/docs/administer/troubleshoot.mdx:
```diff
@@ -108,6 +132,57 @@ To                         Action      From
+## Locked out of admin access
+
+If you are locked out of the admin account for your Firezone instance
+you can recreate your admin user.
```
```suggestion
If the password for the account with email `ADMIN_EMAIL` isn't working, you can reset it using the process below.
```

jamilbk reviewed (changes requested):
Another pass, some more typos etc

jamilbk commented on docs/docs/deploy/omnibus/README.mdx:
```diff
@@ -1,8 +1,13 @@
 sidebar_position: 3
+description:
+  Install Firezone via our Omnibus deployment option to manage
+  secure remote access to private networks and resources.
```
Consistency? Though it's used differently here

```suggestion
  secure access to private networks and resources.
```

Commit:
chore: apply suggestions from code review

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: Jason G <jason@firez.one>

Commit:
chore: Update docs/docs/user-guides/client-instructions.mdx

Co-authored-by: Jamil <jamilbk@users.noreply.github.com>
Signed-off-by: Jason G <jason@firez.one>

Commit:
chore: update title capitalization and other small fixes

Commit:
Merge branch 'docs/general-improvement' of https://github.com/firezone/firezone into docs/general-improvement

jamilbk committed:
Merge branch 'master' into docs/general-improvement

Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk commented on docs/docs/user-guides/add-users.mdx:
```diff
@@ -3,6 +3,8 @@ title: Add Users
 sidebar_position: 1
 ---
 
+# Add and remove users
```
```suggestion
# Add or remove users
```

jamilbk commented on docs/docs/user-guides/client-instructions.mdx:
```diff
@@ -1,14 +1,23 @@
 title: Client Instructions
 sidebar_position: 4
+description:
+    Set up your client app to connect to your organizations private
```
```suggestion
    Set up your client app to connect to your organization's private
```

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx:
```diff
@@ -0,0 +1,87 @@
+
+# Restrict application authentication to VPN users
+
+Some identity providers support restricting SSO functionality
```
```suggestion
```

jamilbk commented on docs/docs/user-guides/use-cases/split-tunnel.mdx:
```diff
@@ -32,10 +37,26 @@ When deciding where to route a packet, Firezone chooses the egress
-users will need to regenerate the configuration files
-and add them to their native WireGuard client.
+Users need to regenerate device configurations to update existing devices
+for the updated configuration changes to `AllowedIPs` and `DNS`.
```
```suggestion
for the updated configuration changes to `AllowedIPs`, `DNS`, `PersistentKeepalive`, `MTU`, and `Endpoint`.
```

jamilbk commented on docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx:
```diff
@@ -0,0 +1,87 @@
+sidebar_position: 1
+description: Configure your identity provider to whitelist traffic from Firezone's WireGuard®-based secure access platform.
+---
+
```
```suggestion
<!-- 
XXX: This document is in draft status. Remove this notice when it's published.
-->
```

jamilbk committed:
Apply suggestions from code review

Signed-off-by: Jamil <jamilbk@users.noreply.github.com>
